lotto.py

```
@@ -0,0 +1,692 @@
+            )
+            await conn.execute(
+                """
+                CREATE INDEX IF NOT EXISTS idx_lotto_nullifiers_round_id
+                ON lotto_nullifiers_spent(round_id)
+                """
+            )
+        except asyncpg.InsufficientPrivilegeError:
+            raise HTTPException(
+                status_code=500,
+                detail="Lotto tables are missing and cannot be created by this DB role",
+            )
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)